localdev/Dockerfile

```diff
@@ -0,0 +1,33 @@
+## This Dockerfile is used for building Horust in a conatiner while developing locally.
+## It follows https://blog.mgattozzi.dev/caching-rust-docker-builds/ AND SHOULD NOT BE USED IN PRODUCTION
+
+FROM rust:1.42 as builder
+WORKDIR /usr/src/myapp
+ARG CARGO_PARAMS
+ARG GIT_COMMIT
+ARG GIT_BRANCH
+ARG IMAGE_NAME
+# The dummy file is just used to get the deps into our project
+# Note that COPY only runs if the files it is provided has changed, otherwise the Docker cache is hit
+COPY localdev/dummy.rs Cargo.toml Cargo.lock ./
+# This uses the dummy Rust program to compile all deps
+RUN sed -i 's^src/main.rs^localdev/dummy.rs^' Cargo.toml
+RUN echo "Running cargo build with dummy program to build all deps: $CARGO_PARAMS" && cargo build $CARGO_PARAMS
+# Now switch back to the main problem before actually creating the Horust binary
+RUN sed -i 's^localdev/dummy.rs^src/main.rs^' Cargo.toml
+# Get everything else into the conatiner
+COPY ../ .
+# And now build without re-compiling all the things! Profit!
+RUN echo "Running cargo build with actual Rust code to build all deps: $CARGO_PARAMS" && cargo build $CARGO_PARAMS
+
+FROM debian:buster-slim
```

Comment:
You probably don't need this second step. I think the ux here should be `make dlocal x`, with x = a command for cargo (so `build`, `build --release`, `test` etc..)

Reply:
So the output is a `horust` binary that's then copied outside of the container into my local machine? Or an image with the compiled binary inside of it?

Reply:
nope nope, this docker container is used just for developing locally, so you probably want to run `cargo test` on the thing in order to build it and making sure tests are passing - so no need to actually build horust. For this reason, I think the ux would be that every parameter you pass to the container, will be passed down to cargo. So you can say `docker [..] test mycoolfeature`.

Reply:
Actually, I think you could even keep this pr open for a while, try to work on the other issue, and improve this pr based on your experience developing with it. As soon as you think the workflow is fine enough, we can merge it. WDYT?

Reply:
Hmmp. Probably a good idea, but that would mean working off of this branch.
That might make another PRs cumbersome to merge later.

What I could do is branch off this branch, work on an issue until I'm satisfied with it, merge this PR and then merge the issue later.

Reply:
I disagree on the merge later thing, this pr adds the dockerfile and updates the makefile, and the pr for the other feature is probably modifying just the rust code, so it should be rather easy to merge later on

Reply:
Alright, let me first get this stabilized and working fast and we'll take it from there

Reply:
Take a look now

Reply:
Irrelevant, resolving